src/hooks/useSearch.ts

```diff
@@ -0,0 +1,79 @@
+import { useState, useRef } from "react";
+import { searchResultSchema } from "@/schemas/searchSchema";
+import { useDebounce } from "./useDebounce";
+import { z } from "zod";
+import { documentSchema } from "@/schemas/searchSchema";
+
+type SearchResult = z.infer<typeof documentSchema>;
+
+export function useSearch() {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [searchResults, setSearchResults] = useState<SearchResult[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const debouncedQuery = useDebounce(searchQuery, 500);
+
+  const previousQuery = useRef(debouncedQuery);
+
+  const performSearch = async (query: string) => {
+    if (!query) {
+      setSearchResults([]);
+      return;
+    }
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      const response = await fetch(
+        "https://danswer.exil-solidaire.fr/api/query/simple-search",
+        {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            query: searchQuery,
+            filters: {
+              source_type: null,
+              document_set: null,
+              time_cutoff: null,
+            },
+          }),
+        }
+      );
+
+      if (!response.ok) {
+        throw new Error(`No results found for "${searchQuery}"`);
+      }
+      const data = await response.json();
+      const results = searchResultSchema.parse(data).documents.map((doc) => ({
+        semantic_identifier: doc.semantic_identifier || "Untitled",
+        link: doc.link,
+      }));
+
+      setSearchResults(results);
+    } catch (err: any) {
```

Comment:
By default all errors in typescript is unknown and it's a better type rather than `any`. Do you know why?
```suggestion
    } catch (err) {
```

Reply:
I had thought about this before. I know that we should avoid using the 'any' type because we won't have control over the object type that we get. But somehow I left it as is. I'll redo it now.